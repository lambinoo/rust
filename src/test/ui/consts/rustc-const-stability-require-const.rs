#![crate_type = "lib"]
#![feature(staged_api)]
#![stable(feature = "foo", since = "1.0.0")]

#[stable(feature = "foo", since = "1.0.0")]
#[rustc_const_unstable(feature = "const_foo", issue = "none")]
pub fn foo() {}
//~^ ERROR attributes `#[rustc_const_unstable]` and `#[rustc_const_stable]` require the function or method to be marked `const`

#[stable(feature = "bar", since = "1.0.0")]
#[rustc_const_stable(feature = "const_bar", since = "1.0.0")]
pub fn bar() {}
//~^ ERROR attributes `#[rustc_const_unstable]` and `#[rustc_const_stable]` require the function or method to be marked `const`

#[stable(feature = "potato", since = "1.0.0")]
pub struct Potato;

impl Potato {
    #[stable(feature = "salad", since = "1.0.0")]
    #[rustc_const_unstable(feature = "const_salad", issue = "none")]
    pub fn salad(&self) -> &'static str { "mmmmmm" }
    //~^ ERROR attributes `#[rustc_const_unstable]` and `#[rustc_const_stable]` require the function or method to be marked `const`

    #[stable(feature = "roasted", since = "1.0.0")]
    #[rustc_const_unstable(feature = "const_roasted", issue = "none")]
    pub fn roasted(&self) -> &'static str { "mmmmmmmmmm" }
    //~^ ERROR attributes `#[rustc_const_unstable]` and `#[rustc_const_stable]` require the function or method to be marked `const`
}

