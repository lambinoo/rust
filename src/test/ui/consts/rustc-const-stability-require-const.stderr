error: attributes `#[rustc_const_unstable]` and `#[rustc_const_stable]` require the function or method to be marked `const`
  --> $DIR/rustc-const-stability-require-const.rs:7:1
   |
LL | #[rustc_const_unstable(feature = "const_foo", issue = "none")]
   | -------------------------------------------------------------- attribute specified here
LL | pub fn foo() {}
   | ^^^^^^^^^^^^ help: make the function or method const

error: attributes `#[rustc_const_unstable]` and `#[rustc_const_stable]` require the function or method to be marked `const`
  --> $DIR/rustc-const-stability-require-const.rs:12:1
   |
LL | #[rustc_const_stable(feature = "const_bar", since = "1.0.0")]
   | ------------------------------------------------------------- attribute specified here
LL | pub fn bar() {}
   | ^^^^^^^^^^^^ help: make the function or method const

error: attributes `#[rustc_const_unstable]` and `#[rustc_const_stable]` require the function or method to be marked `const`
  --> $DIR/rustc-const-stability-require-const.rs:21:5
   |
LL |     #[rustc_const_unstable(feature = "const_salad", issue = "none")]
   |     ---------------------------------------------------------------- attribute specified here
LL |     pub fn salad(&self) -> &'static str { "mmmmmm" }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: make the function or method const

error: attributes `#[rustc_const_unstable]` and `#[rustc_const_stable]` require the function or method to be marked `const`
  --> $DIR/rustc-const-stability-require-const.rs:26:5
   |
LL |     #[rustc_const_unstable(feature = "const_roasted", issue = "none")]
   |     ------------------------------------------------------------------ attribute specified here
LL |     pub fn roasted(&self) -> &'static str { "mmmmmmmmmm" }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: make the function or method const

error: aborting due to 4 previous errors

